src/modules/AddOnProduct.rb

```diff
@@ -404,7 +409,27 @@ def GetBaseProductURL
     #     "username:password@ftp://www.example.org/dir/",
```

Comment:
Not part of your change, but OMG this URI is not what it seems. It does parse, as
```
[[:@scheme, "username"],
 [:@opaque, "password@ftp://www.example.org/dir/"]]
```